lib/fog/compute/kubevirt/models/vminstance.rb

```diff
@@ -14,6 +14,8 @@ class Vminstance < Fog::Model
         attribute :disks,            :aliases => 'spec_disks'
         attribute :volumes,          :aliases => 'spec_volumes'
         attribute :ip_address,       :aliases => 'status_interfaces_ip'
+        attribute :interfaces,       :aliases => 'spec_interfaces'
```

Comment:
@pkliczewski 

both for interfaces and networks there are keys that represent the type of the device (i.e. bridge/slirp for interfaces or pod/multus for networks)
We can either 'normalize' the key into a 'type' element that will be extended with additional supported types as we go or perform no changes to the reported by kubevirt and let clients consume that value as wishes.

In terms of API, it seems that 'nic' class with 'type' attribute will serve this purpose better than a hash.
Save can be applied for networks and disks, in which 'disk' key represents a type, but other keys (floppy/cdrom) will represent different types.

What do you think ? Would we like to have a class for disk/nic/network/volume and normalize the reported data from kubevirt ? Does any of the client perform any assumptions about the existing hash ? (mainly vagrant - I'll check for miq)

Reply:
This is an example which makes [the code](https://github.com/masayag/foreman_kubevirt/commit/9d3d5ce5a3ac9028f33597fdb9be5a2ad18b7549) when not normalizing the data returned by fog-kubevirt:

![selection_878](https://user-images.githubusercontent.com/316242/47618183-c3594a00-dad8-11e8-8d34-4dc652388be9.png)

The disadvantage in that method is the need to constantly release new fog-kubevirt version.

Reply:
I would use a class for all of them. This would help us to hide implementation details. Using a hash would make us dependent on specific kubevirt implementation.

Reply:
@pkliczewski  pls see updated version